classes/Combination.php

```diff
@@ -79,12 +79,10 @@ class CombinationCore extends ObjectModel
         'primary' => 'id_product_attribute',
         'fields' => [
             'id_product' => ['type' => self::TYPE_INT, 'shop' => 'both', 'validate' => 'isUnsignedId', 'required' => true],
-            'location' => ['type' => self::TYPE_STRING, 'validate' => 'isGenericName', 'size' => 64],
```

Comment:
It might be risky to remove those fields right now Since they might be fetched in the front directly from this table via a SQL query (and I don't even mention the modules)

To avoid unexpected behaviour I think it's safer to keep the fields in definition and in the DB, and maintain consistency and synchronization between these values and the one from StockAvailable

We still keep the `loadStockData` to besure the Entity does get the value from StockAvailable but these fields are still kept for backward compatibility and as a safeguard for people who access them directly

Reply:
Also we'd better add allow_null in this definition

Reply:
but what is the point of forcing "loadStockData" in constructor then.
If someone fills quantity/location directly, then he would expect to get those values directly too, so either we don't load stockData in constroctor or we also remove these definitions, it cannot be mixed for read and write :/